lib/directory-tree.js

```diff
@@ -96,13 +96,24 @@ function directoryTree (path, options, onEachFile, onEachDirectory) {
 				item[attribute] = stats[attribute];
 			});
 		}
-		item.children = dirData
-			.map(child => directoryTree(PATH.join(path, child), options, onEachFile, onEachDirectory))
-			.filter(e => !!e);
-		item.size = item.children.reduce((prev, cur) => prev + cur.size, 0);
+		const opts = (options && (options.depth >= 0))
+			? { ...options, depth: Number(options.depth) - 1 }
```

Comment:
Spread operator is only available from Node 8.3, tests are failing because of this :disappointed: